SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosStatusUpdateHandler.java

```diff
@@ -0,0 +1,299 @@
+
+import io.dropwizard.lifecycle.Managed;
+
+@Singleton
+public class SingularityMesosStatusUpdateHandler implements Managed {
+    private static final Logger LOG = LoggerFactory.getLogger(SingularityMesosStatusUpdateHandler.class);
+
+    private final TaskManager taskManager;
+    private final DeployManager deployManager;
+    private final RequestManager requestManager;
+    private final IdTranscoder<SingularityTaskId> taskIdTranscoder;
+    private final SingularityExceptionNotifier exceptionNotifier;
+    private final SingularityHealthchecker healthchecker;
+    private final SingularityNewTaskChecker newTaskChecker;
+    private final SingularitySlaveAndRackManager slaveAndRackManager;
+    private final SingularityLogSupport logSupport;
+    private final SingularityScheduler scheduler;
+    private final Provider<SingularitySchedulerStateCache> stateCacheProvider;
+    private final String serverId;
+    private final BlockingQueue<Protos.TaskStatus> statusUpdateQueue;
+    private final ExecutorService executorService;
+    private Future<?> updateFuture;
+    private final SchedulerDriverSupplier schedulerDriverSupplier;
+    private final AtomicBoolean handlerStarted;
+    private final Lock schedulerLock;
+    private final boolean processStatusUpdatesInSeparateThread;
+
+
+    @Inject
+    public SingularityMesosStatusUpdateHandler(TaskManager taskManager, DeployManager deployManager, RequestManager requestManager,
+        IdTranscoder<SingularityTaskId> taskIdTranscoder, SingularityExceptionNotifier exceptionNotifier, SingularityHealthchecker healthchecker,
+        SingularityNewTaskChecker newTaskChecker, SingularitySlaveAndRackManager slaveAndRackManager, SingularityLogSupport logSupport, SingularityScheduler scheduler,
+        Provider<SingularitySchedulerStateCache> stateCacheProvider, @Named(SingularityMainModule.SERVER_ID_PROPERTY) String serverId,
+        SchedulerDriverSupplier schedulerDriverSupplier,
+        @Named(SingularityMesosModule.SCHEDULER_LOCK_NAME) final Lock schedulerLock,
+        SingularityConfiguration configuration) {
+        this.taskManager = taskManager;
+        this.deployManager = deployManager;
+        this.requestManager = requestManager;
+        this.taskIdTranscoder = taskIdTranscoder;
+        this.exceptionNotifier = exceptionNotifier;
+        this.healthchecker = healthchecker;
+        this.newTaskChecker = newTaskChecker;
+        this.slaveAndRackManager = slaveAndRackManager;
+        this.logSupport = logSupport;
+        this.scheduler = scheduler;
+        this.stateCacheProvider = stateCacheProvider;
+        this.serverId = serverId;
+        this.schedulerDriverSupplier = schedulerDriverSupplier;
+        this.schedulerLock = schedulerLock;
+        this.handlerStarted = new AtomicBoolean();
+
+        this.statusUpdateQueue = new ArrayBlockingQueue<>(configuration.getStatusUpdateQueueCapacity());
+        this.executorService = Executors.newFixedThreadPool(1);
```

Comment:
What about using [this](https://github.com/HubSpot/Singularity/blob/master/SingularityService/src/main/java/com/hubspot/singularity/SingularityManagedScheduledExecutorServiceFactory.java)